### PR TITLE
Sol 677

### PR DESCRIPTION
**Pull request by PhilippeKhalife**
New samples and tutorials using new solace spring auto config projects.

### CODE REVIEW

**Comment by bczoma** [comment]
I have noticed that some new temporary content was not covered by gitignore - fixed that.

File: docs/_tutorials/spring-cloud-autoconf-jms.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+This tutorial is similar to the [Spring Cloud Auto-Config Java]({{ site.baseurl }}/spring-cloud-autoconf-java) tutorial. Like the Spring Cloud Auto-Config Java tutorial, it will introduce you to Solace Messaging for Pivotal Cloud Foundry.  In contrast to the [Spring Cloud Auto-Config Java]({{ site.baseurl }}/spring-cloud-autoconf-java), this application uses the **Solace JMS API** and [solace-jms-spring-boot](https://github.com/SolaceProducts/solace-jms-spring-boot) which in turn is using the Spring Cloud Connectors library and Spring Auto Configuration can auto inject a standard [JMS ConnectionFactory](https://docs.solace.com/API-Developer-Online-Ref-Documentation/jms/javax/jms/ConnectionFactory.html ) directly into your application.
+
+![overview]({{ site.baseurl }}/images/spring-cloud-app-architecture.png){: .center-image}
+
+## Goals
+
+The goal of this tutorial is to demonstrate auto injecting a [JMS ConnectionFactory](https://docs.solace.com/API-Developer-Online-Ref-Documentation/jms/javax/jms/ConnectionFactory.html based on  the application's Cloud Foundry Service Bindings and connect to the Solace Messaging service instance.  This tutorial will show you:
```
Comment (mdspielman) [inline docs/_tutorials/spring-cloud-autoconf-jms.md]: Broken link.

File: docs/_tutorials/spring-cloud-autoconf-jms.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+1. How to Autowire a [JMS ConnectionFactory](https://docs.solace.com/API-Developer-Online-Ref-Documentation/jms/javax/jms/ConnectionFactory.html) into your application
```
Comment (mdspielman) [inline docs/_tutorials/spring-cloud-autoconf-jms.md]: Fix link

File: docs/_tutorials/spring-cloud-autoconf-jms.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+  <artifactId>sol-jms</artifactId>
+  <version>{{ site.jms_version }}</version>
+</dependency>
+```
+
+### Get the API: Using the Solace Developer Portal
+
+The JMS API library can be [downloaded here]({{ site.links-downloads }}){:target="_top"}. The JMS API is distributed as a zip file containing the required jars, API documentation, and examples. 
+
+## Code Walk Through
```
Comment (mdspielman) [inline docs/_tutorials/spring-cloud-autoconf-jms.md]: The "Code Walk Through" title is not a title we use in our samples. This isn't meant to be a code walk through. It's meant to be a tutorial which teaches a concept using code.

File: docs/_tutorials/spring-cloud-autoconf-jms.md [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+    logger.info(String.format(
+            "Discovered Solace Messaging service '%s': HighAvailability? ( %s ), Message VPN ( %s )",
+            discoveredSolaceMessagingService.getId(), discoveredSolaceMessagingService.isHA(),
+            discoveredSolaceMessagingService.getMsgVpnName()));
+}
+```
+
+### Use of Spring JMS in the sample
+
+This Spring Cloud Auto-Config JMS sample app is making use of Spring JMS for messaging. To learn more about Spring JMS, refer to the [Spring JMS Integration documentation](https://docs.spring.io/spring/docs/current/spring-framework-reference/integration.html#jms).
```
Comment (mdspielman) [inline docs/_tutorials/spring-cloud-autoconf-jms.md]: Fix link.

File: docs/_tutorials/spring-cloud-autoconf-jndi.md [hunk 5]
```diff
@@ -0,0 +1,415 @@
+---
+layout: tutorials
+title: Spring Cloud Auto-Config JNDI
+summary: A Simple Spring Cloud Application showing how to consume Solace Messaging as a Service provided by Solace Spring JMS Auto-Configuration using JNDI.
+icon: spring-cloud.png
```
Comment (mdspielman) [inline docs/_tutorials/spring-cloud-autoconf-jndi.md]: Need icons that are unique for each tutorial. We have a graphics designer who can help you if needed.

File: docs/_tutorials/spring-cloud.md [hunk 6]
```diff
@@ -102,7 +102,7 @@ The Pivotal Cloud Foundry environment exposes any bound Service Instances in a J
...
 
-This sample use the [Spring Cloud Connectors](http://cloud.spring.io/spring-cloud-connectors/){:target="_blank"} project to retrieve the `VCAP_SERVICES` information in the form of a `ServiceInfo` object. For Solace Messaging in Cloud Foundry, there is a Spring Cloud Connectors extension in GitHub called [sl-spring-cloud-connectors](https://github.com/SolaceLabs/sl-spring-cloud-connectors). This library parses the `VCAP_SERVICES` environment variable and exposes the information by implementing the `ServiceInfo` interface in the `SolaceMessagingInfo` object.
+This sample use the [Spring Cloud Connectors](http://cloud.spring.io/spring-cloud-connectors/){:target="_blank"} project to retrieve the `VCAP_SERVICES` information in the form of a `ServiceInfo` object. For Solace Messaging in Cloud Foundry, there is a Spring Cloud Connectors extension in GitHub called [sl-spring-cloud-connectors](https://github.com/SolaceProducts/sl-spring-cloud-connectors). This library parses the `VCAP_SERVICES` environment variable and exposes the information by implementing the `ServiceInfo` interface in the `SolaceMessagingInfo` object.
```
Comment (mdspielman) [inline docs/_tutorials/spring-cloud.md]: Fix link to have target.

File: spring-cloud-autoconf-jndi/build.gradle [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+    // Solace Java API & auto-configuration
+    compile("com.solace.spring.boot:solace-jms-spring-boot-starter:${solaceJmsSpringBootVersion}")
+
+    compile("com.solace.spring.boot:solace-jms-spring-boot-autoconfigure:${solaceJmsSpringBootVersion}")
+
+}
+
+task wrapper(type: Wrapper) {
+    gradleVersion = '2.14'
```
Comment (mdspielman) [inline spring-cloud-autoconf-jndi/build.gradle]: Can you pick up a new version of gradle to fix issue with Java 1.9. See other Solace sample for pattern. Current version is 4.3.1

**Comment by mdspielman** [comment]
On the request related to cross linking within the tutorial, we are actually going to update our sample template do to some recent changes to the website structure. So for the PCF tutorials you can update to the new format:

* All links to solace content from a tutorial no longer needs a target. But it must be controlled by a variable with the link being in config.yml.
* All links to external references like GitHub should open in a new tab and therefore should have a target of _blank.

File: docs/_config.yml [hunk 8]
```diff
@@ -10,18 +11,18 @@ jcsmp_version: 10.+
...
 # Links
-docs-core-concepts: http://docs.solace.com/Features/Core-Concepts.htm
-links-downloads: http://dev.solace.com/downloads/
-links-pubsub-tutorial: http://dev.solace.com/get-started/java-tutorials/publish-subscribe_java/
-links-tls-client: http://docs.solace.com/Solace-Messaging-APIs/Creating-Secure-Sessions.htm
+docs-core-concepts: //docs.solace.com/Features/Core-Concepts.htm
+links-downloads: //dev.solace.com/downloads/
+links-pubsub-tutorial: //dev.solace.com/get-started/java-tutorials/publish-subscribe_java/
+links-tls-client: //docs.solace.com/Solace-Messaging-APIs/Creating-Secure-Sessions.htm
 links-tls-server: https://docs.pivotal.io/solace-messaging/installing.html#configure
```
Comment (mdspielman) [inline docs/_config.yml]: Can you remove https or http from all link prefixes.

**Comment by mdspielman** [comment]
One other aspect noted is that the gradle directory is not included in all projects instead of just at the root. The root build is all that should be needed.

**Pull request merged**